### PR TITLE
Event markdown page template

### LINKED ISSUE #30: Feature request: Add events page template
### Is your feature request related to a problem? Please describe.

We currently display event cards on the homepage, but do not have a template for stand-alone pages for these events.

### Describe the solution you'd like

A template is needed to be able to create a subpage for any specific event, which should include:
- Event name, location, dates
- Embedded map (open source option if possible)
- Embedded preview video
- Agenda link / PDF?
- Call-to-action button to register

### Describe alternatives you've considered

(none)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by wackerow**
## Description

Adds markdown template for events pages.

Front matter from markdown files contained within `/src/events` is used to populate event cards, using the following interface (found in `src/interfaces.ts`:

```ts
interface EventFrontmatter {
  title: string
  location: string
  startDate: string
  endDate: string
  imageSrc?: string
  previewLinks?: Link[]
  ctaLinks?: Link[]
  youtube?: string
  coordsOverride?: [Lat, Long]
  mapLabel?: string
}

// where...
interface Link {
  label: string
  href: string
}
type Lat = number
type Long = number
```

### Optional front matter properties for events

- `imageSrc` is the relative path to the image asset to be used as the hero banner.
- `previewLinks` are used to display button links on the event _preview_ cards shown on the homepage. It accepts a list of `Link` objects, each with a `label` and `href`.
- `ctaLinks` are call-to-action button links displayed in the hero and bottom of an event _page_. The first one listed will be styled as a primary solid button; any additional will be styled as a secondary outline button. It accepts a list of `Link` objects, each with a `label` and `href`.
- `youtube` accepts a YouTube video link or video ID, and embeds it below the hero of the event page. It accepts any of the following formats:
  - Standard format: `https://youtube.com/watch?v=1234567890`
  - Embed format: `https://www.youtube.com/embed/1234567890`
  - Shortened format: `https://youtu.be/1234567890`
  - Just the video ID: `1234567890`
- `coordsOverride` can be used to provide a latitude and longitude to override the map location being rendered. See below for more info.
- `mapLabel` can be provided to customize the `<h2>` label shown before an embedded map.

### Location and embedded map

The `location` property is used to display a map on the event page, fetched from [OpenStreetMap](https://www.openstreetmap.org/). If the resulting location is inaccurate or not precise enough, `coordsOverride` can optionally be provided to override this result. If no results are found, the map will not be displayed.

For virtual/remote events, use `location: Remote`, and the map will not be displayed.

_Note: Package `leaflet-geosearch` is being used for geocoding. Using older version `3.6.1` intentionally to avoid the addition of an unnecessary Google dependency added in later versions._

### Event example

```md
---
title: Solidity Summit 2023
location: Istanbul, Turkey
startDate: 2023-11-16
endDate: 2023-11-16
imageSrc: /assets/solidity_summit_2023.png
ctaLinks:
  - label: Speak
    href: https://link.to.speaker.application
  - label: Attend
    href: https://link.to.attendee.application
previewLinks:
  - label: Join us
    href: /event/solidity-summit-2023/
---

Intro text

## First header as h2

...
```

### Event example using `coordsOverride`

```md
---
title: Solidity Summit 2023
location: Istanbul, Turkey
startDate: 2023-11-16
endDate: 2023-11-16
imageSrc: /assets/solidity_summit_2023.png
ctaLinks:
  - label: Speak
    href: https://link.to.speaker.application
  - label: Attend
    href: https://link.to.attendee.application
previewLinks:
  - label: Join us
    href: /event/solidity-summit-2023/
coordsOverride:
  - 41.0082
  - 28.9784
---

Intro text

## First header as h2

...
```

Note that `coordsOverride` is a tuple of two numbers, representing latitude and longitude, respectively. Positive numbers represent north and east, while negative represent south and west.

> The above information has been added to the repo README.md for future reference

## Screenshots
Desktop:
desktop

Mobile:

Notes:
- YouTube video in preview screenshots for demonstration purposes only as a 2023 video is not yet available. This is not included in the PR.
- Speak/Attend buttons do not yet have an `href` property and will be hidden until provided. They are shown in the preview screenshots for demonstration purposes.

## Related issue
- Closes #30

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,10 +131,44 @@ title: Solidity Summit 2023
 location: Istanbul, Turkey
 startDate: 2023-11-16
 endDate: 2023-11-16
-imageSrc: /assets/solidity-summit-2023.png
-links:
+imageSrc: /assets/solidity_summit_2023.png
+ctaLinks:
+  - label: Speak
+    href: https://link.to.speaker.application
```
Comment (NunoFilipeSantos) [inline README.md:137]: @franzihei here.

File: README.md [hunk 2]
```diff
@@ -98,10 +131,44 @@ title: Solidity Summit 2023
...
+  - label: Attend
+    href: https://link.to.attendee.application
```
Comment (NunoFilipeSantos) [inline README.md:139]: And also here @franzihei

File: README.md (hunk 1 above)
Comment (NunoFilipeSantos) [inline README.md:137]:
```suggestion
    href: https://cryptpad.fr/form/#/2/form/view/GMGhskSUobJs5pg5ZZ-l0I75kfVKdvIR0L-B0DSDxRk/
```
@franzihei added.

File: README.md (hunk 2 above)
Comment (NunoFilipeSantos) [inline README.md:139]:
```suggestion
    href: https://cryptpad.fr/form/#/2/form/view/xfCGeeNLIlraxzc46U2fLxAGpbTnlVakiM+18kZJcuI/
```
@franzihei added.

File: src/events/solidity-summit-2023.md [hunk 3]
```diff
@@ -4,7 +4,73 @@ location: Istanbul, Turkey
 startDate: 2023-11-16
 endDate: 2023-11-16
 imageSrc: /assets/solidity_summit_2023.png
-links:
+ctaLinks:
+  - label: Speak
+    href: 
+  - label: Attend
+    href:
```
Comment (NunoFilipeSantos) [inline src/events/solidity-summit-2023.md:11]:
```suggestion
    href: https://cryptpad.fr/form/#/2/form/view/xfCGeeNLIlraxzc46U2fLxAGpbTnlVakiM+18kZJcuI/
```

**Comment by wackerow** [comment]
Thanks for the links @NunoFilipeSantos! Just committed those additions.

New screenshot with how this would look in production:
desktop

On that note, should we update the registration copy?
image

File: src/events/solidity-summit-2023.md [hunk 4]
```diff
@@ -4,7 +4,73 @@ location: Istanbul, Turkey
...
+- Tooling, debuggers and debugging data output
+- Testing frameworks
+- Integration into existing "larger ecosystem tools" (e.g. VSCode via Language Server, LLVM)
+- Source verification
+- Layer 2 support, cross-chain support
+- Mechanisms / programming patterns (on the EVM) like mutexes, state machines, upgradeable contracts, libraries, create2, etc.
+
+## Registrations will open soon.
+
+Tickets for the Solidity Summit 2023 are free and given out on an application basis. Applications will open soon.
```
Comment (wackerow) [inline src/events/solidity-summit-2023.md:73]:
```suggestion
## Registrations are open

Tickets for the Solidity Summit 2023 are free and given out on an application basis. Applications are open now for speakers and attendees.
```
Just a suggestion, feel free to spice it up =)

**Review by NunoFilipeSantos — APPROVED** [review]

File: src/events/solidity-summit-2023.md (hunk 4 above)
Comment (NunoFilipeSantos) [reply src/events/solidity-summit-2023.md:73]: Somehow this comment didn't load. :(

**Comment by NunoFilipeSantos** [comment]
@wackerow I'll do a separate PR for the copy now. How long until we see the new design live? :D

**Pull request merged**